HPXMLtoOpenStudio/resources/hvac_sizing.rb

```diff
@@ -4290,32 +4295,41 @@ def self.calc_slab_f_value(slab, ground_conductivity)
       u_effective = []
       for radius in 0..path_radius
         spl = [Math::PI * radius - 1, 0].max # soil path length (SPL)
-
+        r_ins = 0.0
         # Concrete, gravel, and insulation
         if radius == 0
           r_concrete = 0.0
           r_gravel = 0.0 # No gravel on edge
           if slab.perimeter_insulation_depth > 0
             r_ins = slab.perimeter_insulation_r_value # Insulation on edge
-          else
-            r_ins = 0.0
           end
         else
           r_concrete = Material.Concrete(slab.thickness).rvalue
           r_gravel = [slab_r_gravel_per_inch * (12.0 - slab.thickness), 0].max
           if slab.under_slab_insulation_spans_entire_slab
-            r_ins = slab.under_slab_insulation_r_value
-          elsif radius <= slab.under_slab_insulation_width && radius <= slab.perimeter_insulation_depth
-            r_ins = slab.under_slab_insulation_r_value + slab.perimeter_insulation_r_value
-          elsif radius <= slab.under_slab_insulation_width
-            r_ins = slab.under_slab_insulation_r_value
-          elsif radius <= slab.perimeter_insulation_depth
-            r_ins = slab.perimeter_insulation_r_value
-          else
-            r_ins = 0.0
+            r_ins += slab.under_slab_insulation_r_value.to_f
+          end
+          if radius <= slab.under_slab_insulation_width.to_f && radius <= slab.perimeter_insulation_depth.to_f
+            r_ins += slab.under_slab_insulation_r_value.to_f + slab.perimeter_insulation_r_value.to_f
+          end
+          if radius <= slab.under_slab_insulation_width.to_f
+            r_ins += slab.under_slab_insulation_r_value.to_f
+          end
+          if radius <= slab.perimeter_insulation_depth.to_f
+            r_ins += slab.perimeter_insulation_r_value.to_f
+          end
+          if slab.exterior_horizontal_insulation_r_value.to_f > 0
```

Comment:
I just realized that we default all the slab insulation properties when they are omitted, so we shouldn't need to do `.to_f` everywhere here. I wonder why the code was hitting errors without that. I'm going to try reverting those changes and, if there's an error, look at it more closely.